volatility3/framework/plugins/windows/mftscan.py

```diff
@@ -174,126 +190,120 @@ def run(self):
         )
 
 
-class ADS(interfaces.plugins.PluginInterface):
+class ADS(MFTScan):
     """Scans for Alternate Data Stream"""
 
     _required_framework_version = (2, 0, 0)
 
-    @classmethod
-    def get_requirements(cls):
-        return [
-            requirements.TranslationLayerRequirement(
-                name="primary",
-                description="Memory layer for the kernel",
-                architectures=["Intel32", "Intel64"],
-            ),
-            requirements.VersionRequirement(
-                name="yarascanner", component=yarascan.YaraScanner, version=(2, 0, 0)
-            ),
-        ]
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        # which DATA attribute should be displayed
+        self._display_first_data = False
+
+    def _parse_data_record(self, mft_record, attr):
```

Comment:
Could we add typing information to this just to make it easier to spot things going wrong please?  This will return a `Generator` type so the return at the start wouldn't require `Optional` in the type.  Makes it easier to figure out if code using it will throw errors...

Reply:
fixed